navigation/insilico_rotater.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+namespace BoBRobotics {
+namespace Navigation {
+//------------------------------------------------------------------------
+// BoBRobotics::Navigation::InSilicoRotater
+//------------------------------------------------------------------------
+/*!
+ * \brief A "rotater" for PerfectMemoryBase classes which rotates a panoramic
+ *        image by moving columns of pixels from one side of the image to the
+ *        other
+ */
+class InSilicoRotater
+{
+public:
+    InSilicoRotater(const cv::Size &unwrapRes,
+                    const cv::Mat &maskImage,
+                    const cv::Mat &image,
+                    const unsigned int scanStep = 1)
+      : m_ScanStep(scanStep)
+    {
+        assert(image.cols == unwrapRes.width);
+        assert(image.rows == unwrapRes.height);
+        assert(image.type() == CV_8UC1);
+
+        image.copyTo(m_Image);
+        maskImage.copyTo(m_MaskImage);
+    }
+
+    template<class Func>
+    void operator()(Func func)
+    {
+        size_t i = 0;
+        while (true) {
+            func(m_Image, m_MaskImage, i);
+
+            i += m_ScanStep;
+            if (i >= (size_t) m_Image.cols) {
```

Comment:
Was there a particular reason this wasn't a ``for(size_t i = 0; i < (size_t)m_Image.cols; i += m_ScanStep)``? (It may well have come from my code originally  :smile: )

Reply:
:+1: past-somewhat was smort